File: java/chaincue-real-estate/src/main/java/com/chaincuerealestate/chaincuerealestate/routes/houses/HousesPageDTO.java
package com.chaincuerealestate.chaincuerealestate.routes.houses;

public record HousesPageDTO(
        Country[] countries,
        House[] houses
) {

    record Country(
            String name
    ) {
    }

    record House(
            String id,
            String title,
            String location,
            int numberRooms,
            int beds,
            String dollarPrice,
            String cryptoPrice,
            String src
    ) {
    }
}

File: kotlin/chaincue-real-estate/src/main/kotlin/com/example/chaincuerealestate/routes/houses/HousesPageDTO.kt
package com.example.chaincuerealestate.routes.houses

class HousesPageDTO(
    val countries: Array<Country>,
    val houses: Array<House>
) {
    data class Country(
        val name: String
    )

    data class House(
        val id: String,
        val title: String,
        val location: String,
        val numberRooms: Int,
        val beds: Int,
        val dollarPrice: String,
        val cryptoPrice: String,
        val src: String
    )
}
